File: 2023/java/src/test/java/org/frieder/aoc/day4/TestDay4A.java
package org.frieder.aoc.day4;

import org.frieder.aoc.day4.a.Solution4A;
import org.junit.Test;

import java.io.IOException;

import static org.junit.Assert.assertEquals;

public class TestDay4A
{
    @Test
    public void testExample() throws IOException {
        assertEquals(13, Solution4A.getResult("../resources/4/test_a.txt"));
    }

    @Test
    public void testResult() throws IOException {
        assertEquals(23235, Solution4A.getResult("../resources/4/input.txt"));
    }
}

File: 2023/php/test/TestDay4A.php
<?php declare(strict_types=1);

$loader = require __DIR__ . '/../vendor/autoload.php';

use day4\a\Solution4A;
use PHPUnit\Framework\TestCase;
use utils\TestLogger;

final class TestDay4A extends TestCase
{

    public function testExample(): void
    {
        $this->assertSame(13, Solution4A::getResult("../resources/4/test_a.txt", TestLogger::createTestLogger()));
    }

    public function testResult(): void
    {
        $this->assertSame(23235, Solution4A::getResult("../resources/4/input.txt", TestLogger::createTestLogger()));
    }

}